src/pybind/mgr/cephadm/serve.py

```diff
@@ -1091,10 +1091,11 @@ def _check_daemons(self) -> None:
                 self.log.debug(f'{dd.name()} deps {last_deps} -> {deps}')
                 self.log.info(f'Reconfiguring {dd.name()} (dependencies changed)...')
                 action = 'reconfig'
-                # we need only redeploy if secure_monitoring_stack value has changed:
+                # we need only redeploy if secure_monitoring_stack or mgmt-gateway value has changed:
+                # TODO(redo): check if we should just go always with redeploy (it's fast enough)
                 if dd.daemon_type in ['prometheus', 'node-exporter', 'alertmanager']:
-                    diff = list(set(last_deps) - set(deps))
-                    if any('secure_monitoring_stack' in e for e in diff):
+                    diff = list(set(last_deps).symmetric_difference(set(deps)))
+                    if any(('secure_monitoring_stack' in e) or ('mgmt-gateway' in e) for e in diff):
```

Comment:
Rather than hard-coding individual exceptions, it's better to replace that with a generic approach:
```suggestion
                    REDEPLOY_SERVICES = ['secure_monitoring_stack', 'mgmt-gateway']
                    if any(svc for svc in REDEPLOY_SERVICES in e for e in diff):
```